src/test/java/seedu/address/logic/commands/SetDefaultGroupCommandTest.java

```diff
@@ -0,0 +1,316 @@
+import seedu.address.model.VersionedContents;
+import seedu.address.model.person.Mod;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.UniqueModuleList;
+import seedu.address.model.tasks.ReadOnlyTaskList;
+import seedu.address.model.tasks.Task;
+import seedu.address.testutil.ModuleBuilder;
+
+public class SetDefaultGroupCommandTest {
+
+    @Test
+    public void constructor_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new SetDefaultGroupCommand(null));
+    }
+
+    @Test
+    public void execute_moduleAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingModAdded modelStub = new ModelStubAcceptingModAdded();
+        Mod validmod = new ModuleBuilder().build();
+
+        CommandResult commandResult = new SetDefaultGroupCommand(validmod).execute(modelStub);
+
+        assertEquals(String.format(SetDefaultGroupCommand.MESSAGE_SUCCESS, validmod.value,
+                    validmod.getDefaultGroup()), commandResult.getFeedbackToUser());
+
+        UniqueModuleList tester = new UniqueModuleList();
+        tester.add(validmod);
+        assertEquals(tester, modelStub.moduleList);
+
+        validmod.setDefaultGroup("T21");
+        CommandResult commandResult1 = new SetDefaultGroupCommand(validmod).execute(modelStub);
+        assertEquals(String.format(SetDefaultGroupCommand.MESSAGE_DEFAULT_UPDATE, validmod.value, "T21",
+                    validmod.getDefaultGroup()), commandResult1.getFeedbackToUser());
+        UniqueModuleList tester1 = new UniqueModuleList();
+        tester1.add(validmod);
+        assertEquals(tester1, modelStub.moduleList);
+
+        Mod validmod2 = new Mod("CS3232");
+        CommandResult commandResult2 = new SetDefaultGroupCommand(validmod2, "T12").execute(modelStub);
+        assertEquals(String.format(SetDefaultGroupCommand.MESSAGE_SUCCESS, validmod2.value,
+                    validmod2.getDefaultGroup()), commandResult2.getFeedbackToUser());
+    }
+
+    @Test
+    public void equals() {
+        Mod aliceMod = new ModuleBuilder().withMod("CS2100").withGroup("T01").build();
+        Mod bobMod = new ModuleBuilder().withMod("CS2108").withGroup("T02").build();
+        SetDefaultGroupCommand addAliceDefaultGroup = new SetDefaultGroupCommand(aliceMod);
+        SetDefaultGroupCommand addBobDefaultGroup = new SetDefaultGroupCommand(bobMod);
+
+        // same object -> returns true
+        assertTrue(addAliceDefaultGroup.equals(addAliceDefaultGroup));
+
+        // same values -> returns true
+        SetDefaultGroupCommand addAliceCommandCopy = new SetDefaultGroupCommand(aliceMod);
+        assertTrue(addAliceDefaultGroup.equals(addAliceCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addAliceDefaultGroup.equals(1));
+
+        // null -> returns false
+        assertFalse(addAliceDefaultGroup.equals(null));
+
+        // different person -> returns false
```

Comment:
Different commands

Reply:
wdym?

Reply:
Should this be referring to two different instances of the SetDefaultGroupCommand instead of having different people?